sljit_src/sljitConfigInternal.h

```diff
@@ -122,6 +105,27 @@ extern "C" {
 #error "Multiple architectures are selected"
 #endif
 
+#if !((defined SLJIT_CONFIG_X86_32 && SLJIT_CONFIG_X86_32) \
+	|| (defined SLJIT_CONFIG_X86_64 && SLJIT_CONFIG_X86_64) \
+	|| (defined SLJIT_CONFIG_ARM_V5 && SLJIT_CONFIG_ARM_V5) \
+	|| (defined SLJIT_CONFIG_ARM_V7 && SLJIT_CONFIG_ARM_V7) \
+	|| (defined SLJIT_CONFIG_ARM_THUMB2 && SLJIT_CONFIG_ARM_THUMB2) \
+	|| (defined SLJIT_CONFIG_ARM_64 && SLJIT_CONFIG_ARM_64) \
+	|| (defined SLJIT_CONFIG_PPC_32 && SLJIT_CONFIG_PPC_32) \
+	|| (defined SLJIT_CONFIG_PPC_64 && SLJIT_CONFIG_PPC_64) \
+	|| (defined SLJIT_CONFIG_MIPS_32 && SLJIT_CONFIG_MIPS_32) \
+	|| (defined SLJIT_CONFIG_MIPS_64 && SLJIT_CONFIG_MIPS_64) \
+	|| (defined SLJIT_CONFIG_SPARC_32 && SLJIT_CONFIG_SPARC_32) \
+	|| (defined SLJIT_CONFIG_TILEGX && SLJIT_CONFIG_TILEGX) \
+	|| (defined SLJIT_CONFIG_UNSUPPORTED && SLJIT_CONFIG_UNSUPPORTED) \
+	|| (defined SLJIT_CONFIG_AUTO && SLJIT_CONFIG_AUTO) \
```

Comment:
Can the last two conditions be merged?
`|| (!defined SLJIT_CONFIG_AUTO || !SLJIT_CONFIG_AUTO)`

Reply:
note the whole condition is !() so the first one means SLJIT_CONFIG_AUTO is defined and 1 (actually, non zero), and the second one means SLJIT_CONFIG_AUTO is undefined.

you are correct that the whole "defined" check is irrelevant but it might be better to just replace it with ") || !SLJIT_CONFIG_AUTO"

in the same line we could get rid of all those other "defined" checks and rely on the C99 rule §6.10.1 ¶3 that states all undefined macros default to 0, or invert the logic and only care about any of them being defined.

Reply:
a simpler (an not redundant for SLJIT_CONFIG_AUTO) conditional in last revision, but note that it misconfigures the code (or maybe configures it correctly but that breaks the build) for at least SLJIT_CONFIG_UNSUPPORTED (hence why in draft)

FWIW, it might seem that with the previous logic some of the SLJIT_CONFIG_X86_64 were still being used when SLJIT_CONFIGU_UNSUPPORTED was defined